introduced BetonQuestLoggerFactory to be able to inject a BetonQuestLogger (#2447)
BetonQuest/BetonQuest: docs/API/Logging.md

this should be the case. One should always initialize the logger in the 
main class and pass it to the other classes. 

!!! note ""
    === "For Addons: Using the ServiceManager"
        This is the recommended way to obtain a BetonQuestLoggerFactory instance. Simply add this in your `onEnable()` method:        
        ````java linenums="1"
        final BetonQuestLoggerFactory loggerFactory = Bukkit.getServicesManager().load(BetonQuestLoggerFactory.class);
        ````
        
        This will only return a BetonQuestLoggerFactory instance if BetonQuest is installed and already loaded.
        Otherwise it will return `null`.
        
    === "For internal development: Legacy classes"
        If you are working on a legacy class that requires a `BetonQuestLogger` and exists in an architecture that has not
        been migrated to pass the `BetonQuestLoggerFactory` around, you can obtain the `BetonQuestLoggerService` like so:
        
        ````java linenums="1"
        final BetonQuestLoggerFactory loggerFactory = BetonQuest.getInstance().getLoggerFactory();
        ````

>         `BetonQuest.getInstance()` will return `null` if BetonQuest is not installed or not loaded yet.

Author: ```suggestion
        `BetonQuest.getInstance()` will only return a BetonQuest instance if BetonQuest is already enabled
```

            
            ````java linenums="1"
            final BetonQuestLogger logger = loggerFactory.create(MyClass.class, "MyCustomTopic");
            ````
    
    !!! warning "Get the logger in your JavaPlugin class"
        The methods described above don't work for your plugin's main class (or any other class that extends `Plugin`). 
        Create the logger instance in the `onEnable()` method instead like this:
    
        === "Without topic"
            ````java linenums="1"
            public final class BetonQuestAddon extends JavaPlugin {
        
                private static BetonQuestLogger log;
        
                @Override
                public void onEnable() {
                    log = BetonQuestLogger.create(this);
            ````
    
        === "With topic"
            ````java linenums="1"
            public final class BetonQuestAddon extends JavaPlugin {
        
                private static BetonQuestLogger log;
        
                @Override
                public void onEnable() {
                    log = BetonQuestLogger.create(this, "MyCustomTopic");
            ````

## Using the BetonQuestLogger

> A BetonQuestLogger will be available as the variable `LOG` once you [obtained a BetonQuestLogger instance](#obtaining-a-betonquestlogger-instance). 

Reviewer: This part still assumes a static `BetonQuestLogger` instance stored in the field `LOG`.

Remove this line and change the following line to:
```markdown
The BetonQuestLogger has a bunch of methods for all use cases. Its JavaDocs explain when and how to use these.
```
A suggestion was no option as this text is out of scope of the changes in GitHub.

## Obtaining a BetonQuestLogger Instance

First you need to obtain a `BetonQuestLoggerFactory` instance. For this you have two ways:

!!! note ""

    === "Using the ServiceManager"
        !!! abstract ""
            This is the recommended way to obtain a BetonQuestLoggerFactory instance. Simply add this in your `onEnable()` method:
            
            ````java linenums="1"
            final BetonQuestLoggerFactory loggerFactory = Bukkit.getServicesManager().load(BetonQuestLoggerFactory.class);
            ````
            
            This only works if BetonQuest is installed on the server. If you want to support servers without BetonQuest,
            you need to do it this way:
            
            ````java linenums="1"
            RegisteredServiceProvider<BetonQuestLoggerFactory> provider = Bukkit.getServicesManager().getRegistration(BetonQuestLoggerFactory.class);
            if (provider != null) {
                BetonQuestLoggerFactory loggerFactory = provider.getProvider();

>             }

Reviewer: This example will fail with a ClassNotFoundException, as the ClassLoader will try to load `BetonQuestLoggerFactory` which does not exist either.

Also, it is not required, as `.getServicesManager().load(BetonQuestLoggerFactory.class)` will just return null if there is no instance provided. But that can only happen if the plugin loads before BetonQuest.

            ````
    === "Using the BetonQuest instance"
        !!! abstract ""
            This can be used but we do not recommend it, as we may change the way the BetonQuest instance is obtained in the future.
            
            ````java linenums="1"
            final BetonQuestLoggerFactory loggerFactory = BetonQuest.getInstance().getLoggerFactory();
            ````
    
    ??? info "`BetonQuestLoggerFactory` additional background implementation information"
        As the BetonQuestLoggerFactory is a service, it is not guaranteed that the instance you get
        is the one BetonQuest created by default. But here we expain the behavior of the default BetonQuestLoggerFactory.
        
        First there is the `DefaultBetonQuestLoggerFactory` class,
        which is the default implementation of the `BetonQuestLoggerFactory` interface.

>         It simnply creates child logger for the given class for the Logger of your plugin.

Reviewer: Typo and grammar.
```suggestion
        It simply creates a child logger for the given class using the Logger of your plugin.
```
